File: generators/HttpRequestHeaderCodeGenerator/Templates/root.template.swift
/**
 Http Request Header
 */
enum HttpRequestHeader: String {

    /// Returns the element matching the given value.
    public static func parse(value: String) -> HttpRequestHeader {
        return HttpRequestHeader.init(rawValue: value)!!
    }

    /// Returns the element matching the given value or null.
    public static func parseOrNull(value: String) -> HttpRequestHeader? {
        return HttpRequestHeader.init(rawValue: value)
    }


%%ITEMS%%
}

File: generators/HttpRequestHeaderCodeGenerator/Templates/root.template.kt
package your.package

/**
 * HTTP Request Header
 */
enum class HttpRequestHeader(val value: String) {

%%ITEMS%%
    ;


    companion object {

        /**
         * Returns the element matching the given value.
         */
        @JvmStatic
        fun parse(value: String): HttpRequestHeader {
            return values().single { it.value == value }
        }

        /**
         * Returns the element matching the given value or null.
         */
        @JvmStatic
        fun parseOrNull(value: String): HttpRequestHeader? {
            return values().singleOrNull { it.value == value }
        }
    }
}
